renku/core/commands/move.py

```diff
@@ -19,101 +19,165 @@
-                dataset.rename_files(lambda key: renames.get(key, key))
-
-                dataset.to_yaml()
-            communication.update_progress(progress_text, amount=1)
-    finally:
-        communication.finalize_progress(progress_text)
-
-    # 3. Manage .gitattributes for external storage.
-    tracked = tuple()
-    if client.check_external_storage():
-        tracked = tuple(path for path, attr in client.find_attr(*files).items() if attr.get("filter") == "lfs")
-        client.untrack_paths_from_storage(*tracked)
-
-        if client.find_attr(*tracked):
-            communication.warn("There are custom .gitattributes.\n")
-            if communication.confirm('Do you want to edit ".gitattributes" now?'):
-                edit_command(filename=str(client.path / ".gitattributes"))
-
-        if tracked:
-            lfs_paths = client.track_paths_in_storage(*(destinations[path] for path in tracked))
-            show_message = client.get_value("renku", "show_lfs_message")
-            if lfs_paths and (show_message is None or show_message == "True"):
-                communication.info(
-                    "Adding these files to Git LFS:\n"
-                    + "\t{}".format("\n\t".join(lfs_paths))
-                    + "\nTo disable this message in the future, run:"
-                    + "\n\trenku config set show_lfs_message False"
-                )
-
-    # 4. Handle symlinks.
-    dst.parent.mkdir(parents=True, exist_ok=True)
-
-    for source, target in destinations.items():
-        src = Path(source)
-        if src.is_symlink():
-            Path(target).parent.mkdir(parents=True, exist_ok=True)
-            Path(target).symlink_to(os.path.relpath(str(src.resolve()), start=os.path.dirname(target)))
-            src.unlink()
-            del files[source]
-
-    # Finally move the files.
-    final_sources = list(set(files.values()))
-    if final_sources:
-        run(["git", "mv"] + final_sources + [destination], check=True)
+        client.repo.git.mv([str(src) for src in sources] + [str(destination)], force=force)
+    except git.GitError as e:
+        raise errors.OperationError(f"Git operation failed: {e}")
 
+    # Handle symlinks
+    for src, dst in files.items():
+        if dst.is_symlink():
+            target = src.parent / os.readlink(dst)
+            dst.unlink()
+            Path(dst).symlink_to(os.path.relpath(target, start=os.path.dirname(dst)))
 
-def move_command():
-    """Command to move or rename a file, a directory, or a symlink."""
-    return Command().command(_move).require_migration().require_clean().with_commit()
+    files_to_untrack = (str(src.relative_to(client.path)) for src in files)
+    client.untrack_paths_from_storage(*files_to_untrack)
+    # NOTE: Warn about filter after untracking from LFS to avoid warning about LFS filters
+    _warn_about_git_filters(client, files)
+    client.track_paths_in_storage(*[dst for dst in files.values() if not dst.is_dir()])
```

Comment:
I'm not sure tbh. We cannot do that all the time since there might be files in the sources that are not tracked in LFS and need to be excluded here. I believe this is the least error-prone way of tracking files.